library/std/src/process.rs

```diff
@@ -2051,6 +2052,15 @@ impl<E: fmt::Debug> Termination for Result<!, E> {
     }
 }
 
+#[unstable(feature = "termination_trait_lib", issue = "43301")]
+impl<E: fmt::Debug> Termination for Result<Infallible, E> {
+    fn report(self) -> i32 {
+        let Err(err) = self;
+        eprintln!("Error: {:?}", err);
+        ExitCode::FAILURE.report()
```

Comment:
This could forward to `Termination` implementation for `Result<!, E>` instead of duplicating code.

```suggestion
        Err::<!, _>(err).report()
```